drivers/pcie/endpoint/pcie_ep_bcm_iproc.c

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright 2020 Broadcom
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/pcie/endpoint/pcie_ep.h>
+
+#define LOG_LEVEL CONFIG_PCIE_EP_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(iproc_pcie);
+
+#include "pcie_ep_bcm_iproc.h"
+#include "pcie_ep_bcm_iproc_regs.h"
+
+#define DT_DRV_COMPAT brcm_iproc_pcie_ep
+
+/* Helper macro to read 64-bit data using two 32-bit data read */
+#define sys_read64(addr)    (((uint64_t)(sys_read32(addr + 4)) << 32) | \
+			     sys_read32(addr))
+
+static int iproc_pcie_conf_read(struct device *dev, uint32_t offset,
+				uint32_t *data)
+{
+	const struct iproc_pcie_ep_config *cfg = dev->config_info;
+
+	/* Write offset to Configuration Indirect Address register */
+	pcie_write32(offset, &cfg->base->paxb_config_ind_addr);
+
+	/* Read data from Configuration Indirect Data register */
+	*data = pcie_read32(&cfg->base->paxb_config_ind_data);
+
+	return 0;
+}
+
+static void iproc_pcie_conf_write(struct device *dev, uint32_t offset,
+				  uint32_t data)
+{
+	const struct iproc_pcie_ep_config *cfg = dev->config_info;
+
+	/* Write offset to Configuration Indirect Address register */
+	pcie_write32(offset, &cfg->base->paxb_config_ind_addr);
+
+	/* Write data to Configuration Indirect Data register */
+	pcie_write32(data, &cfg->base->paxb_config_ind_data);
+}
+
+static int iproc_pcie_map_addr(struct device *dev, uint64_t pcie_addr,
+			       uint64_t *mapped_addr, uint32_t size,
```

Comment:
why not use `size_t` for the size?

Reply:
I looked into this. `size_t` is 64-bit wide when building zephyr with LP64 and is 32-bit wide with ILP32. We really want this variable to be just 32-bit wide. So, I would like to stick to `uint32_t`.